internal/server/openapi.go

```diff
@@ -122,71 +116,51 @@ func getFuncName(i interface{}) string {
 	return name
 }
 
-func orderedTagNames() []string {
-	tagNames := make([]string, 0, len(funcPartialNameToTagNames))
-	for k := range funcPartialNameToTagNames {
-		tagNames = append(tagNames, k)
-	}
-
-	sort.Strings(tagNames)
-	return tagNames
-}
-
-// createComponent creates and returns the SchemaRef for a type. If the type is
-// a struct, the definition of the struct is added  to the schemas map.
+// createComponent creates and returns the SchemaRef for a response type.
 func createComponent(schemas openapi3.Schemas, rst reflect.Type) *openapi3.SchemaRef {
-	//nolint:exhaustive
-	switch rst.Kind() {
-	case reflect.Pointer:
-		return createComponent(schemas, rst.Elem())
-	case reflect.Slice:
```

Comment:
This is because slices will be contained within the 'items' key, correct?

Reply:
Exactly! Hopefully all list endpoints will use that generic type, and if we have others it's generally a good idea to return a struct instead of a slice so that we can add additional fields without needing to break compatibility.